src/generators/php/presets/JsonSerializablePreset.ts

```diff
@@ -0,0 +1,88 @@
+import { PhpPreset } from '../PhpPreset';
+import { PhpRenderer } from '../PhpRenderer';
+import { ConstrainedMetaModel } from '../../../models';
+
+function renderSelf({
+  content,
+  renderer
+}: {
+  content: string;
+  renderer: PhpRenderer<ConstrainedMetaModel>;
+}): string {
+  renderer.dependencyManager.addDependency('use JsonSerializable;');
+
+  const contentLines = content.split('\n');
+  contentLines[0] += ` implements JsonSerializable`;
```

Comment:
Implementing adding interfaces to classes this may cause incompatibilities when multiple presets try to implement interfaces. Can you imagine any other solution here?

Reply:
Uhh, yeah, this one is tough. 

We could later on when the problem occurs implement a support function to add interfaces.

Alternatively, all functionality that need to add a new interface would need to be part of the same preset and then we control what is rendered based on options. But that's not really maintainable long term 🤔

Reply:
I am fine with postponing the problem. I see no other use-case at the moment. Likewise, I was also considering putting it somehow to the options.

I will close this conversation.